Feature/three address code

yanushka opened the pull request.

yanushka committed:
Adding interface for representing ThreeAddressCode

yanushka committed:
Making interface public

slaffka93 committed:
BaseBlock init

yanushka committed:
Adding classes for representation three address code

yanushka committed:
Merge branch 'feature/three-address-code' of https://github.com/mmcsOptimizingCompilers2016/pile.git

yanushka committed:
Changing BaseThreeAddressCode

yanushka committed:
Adding interface for representing values in three-address-code

geo2a commented on OptimizingCompilers2016/Library/Library.csproj:
```diff
@@ -33,6 +33,8 @@
     <Reference Include="System" />
     <Reference Include="System.Core" />
+    <Reference Include="System.Drawing" />
+    <Reference Include="System.Windows.Forms" />
```
Как зависимости от Windows.Forms и System.Drawing попали в Library? Это, конечно, незначительная деталь, но мой внутренний перфекционист не позволяет мне смёржить этот пуллреквест. Давайте завтра их уберём и тогда смёржим.

elena-gordienko committed:
fix typo in class name

elena-gordienko committed:
switch to the new representation of basic classes

elena-gordienko committed:
WIP: remove old basic classes

skapix committed:
Fix some defects

skapix committed:
Return of the RepUntNode

yanushka committed:
1. Removing unused references.
2. Changing IValue interface and its implementation.
3. Adding base class for indetificator and label values
